streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableSourceTest.java

```diff
@@ -24,19 +24,34 @@
 import org.apache.kafka.streams.kstream.KTable;
 import org.apache.kafka.test.KStreamTestDriver;
 import org.apache.kafka.test.MockProcessorSupplier;
+import org.junit.After;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 import java.io.File;
 import java.io.IOException;
-import java.nio.file.Files;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
 public class KTableSourceTest {
 
-    final private Serde<String> stringSerde = new Serdes.StringSerde();
+    final private Serde<String> stringSerde = Serdes.String();
+
+    private KStreamTestDriver driver = null;
+
+    @After
+    public void cleanup() {
+        if (driver != null) {
+            driver.close();
+        }
+        driver = null;
+    }
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
```

Comment:
So, I took a look at the implementation of this and it's kind of weird because you lose the atomicity guarantee:

```java
    private File createTemporaryFolderIn(File parentFolder) throws IOException {
        File createdFolder = File.createTempFile("junit", "", parentFolder);
        createdFolder.delete();
        createdFolder.mkdir();
        return createdFolder;
    }
```

The obvious way to do this is to use `Files.createTempDirectory` (which is what we do in `TestUtils.tempRelativeDir`), but maybe they didn't want to require Java 7.